### DISCUSSION TITLE
different result from PED-ANOVA vs fANOVA(default evaluator)

### DISCUSSION THREAD

**Discussion by GhsJahanmir** [optuna/optuna#5695] [Q&A]
Dear All, 

I have another question that I’m still trying to resolve.

Following the suggestions from the discussion [here](https://github.com/optuna/optuna/discussions/5680), I utilized both evaluators in Optuna to determine which hyperparameters are more important (I did not fix the seed in the default evaluator, but if I increase the number of trials, the values tend to converge). However, they yield different and somewhat contradictory results. Below are typical results I've obtained from several studies.

![image](https://github.com/user-attachments/assets/c7a2eab1-60aa-4496-9b5c-582f4b6be51d)

I understand that these two methods provide different insights, but I’m unsure which one I should rely on to identify the most important hyperparameters.

Additionally, I reviewed the underlying paper on PED, but I'm still unclear about what "local" versus "general" means in the context of parameter importance.

Thank you in advance for your assistance.

Best regards,
Ghs

**Comment by XinY-Z** [comment]
You might find some insights in here: https://medium.com/@mdshah930/understranding-hyperparameter-importance-in-optuna-mastering-optuna-part-2-c5c88956152a

**Comment by nabenabe0928** [comment]
Hey @GhsJahanmir ,

Sorry for the late response:(
I simply did not notice your thread.
Ok, so I am the PED-ANOVA paper author.
I have [a slide](https://drive.google.com/file/d/1pwCa0fgh6jIooW0admOmkApa8VQqAgTF/view) that intuitively explains PED-ANOVA.

The somewhat contradictory results are indeed intended.
They come from the different philosophies and they do regard different parameters as important.

Simply put, f-ANOVA says a parameter is important when the parameter may completely devastate the objective.
For example, when you set a range of learning rate to [-1, 1], a negative learning rate obviously fuck up the whole learning process, so the performance variation caused by learning rate is quite huge, leading to learning rate to be an important parameter.

Meanwhile, PED-ANOVA may not consider such a parameter as important.
PED-ANOVA, in a way, discards the bad information and genuinely focuses on good information, meaning that a parameter is considered to be good if performant parameter values are concentrated in a certain sub-range.

Let me rephrase a bit more and summarize what I meant:
- Use f-ANOVA if you are interested in which parameter may fuck up your objective function
- Use PED-ANOVA if you are interested in which parameter may impact the very last performance improvement
